native/cocos/renderer/pipeline/custom/details/SerializationUtils.h

```diff
@@ -62,13 +62,21 @@ void save(OutputArchive& ar, const T& v) {
 
 template <class T,
           std::enable_if_t<
-              std::is_enum_v<T> || std::is_integral_v<T> || std::is_floating_point_v<T>,
+              std::is_integral_v<T> || std::is_floating_point_v<T>,
               bool> = false>
 void load(InputArchive& ar, T& v) {
-    static_assert(!std::is_enum_v<T> || sizeof(T) <= sizeof(uint32_t)); // enum can only be 1, 2, 4 bytes
+    static_assert(sizeof(T) <= sizeof(uint32_t)); // enum can only be 1, 2, 4 bytes
     v = static_cast<T>(ar.readNumber());
 }
 
+template <class T,
+          std::enable_if_t<std::is_enum_v<T>,
+              bool> = false>
+void load(InputArchive& ar, T& v) {
+    v = static_cast<T>(static_cast<typename std::underlying_type<T>::type>(ar.readNumber()));
+}
```

Comment:
Why need special operation for enum?

Reply:
Cast from double to enum might not be supported.

Reply:
K, then please add some comment.

Reply:
K, then please add some comment.